pkg/controllers/tagging/tagging_controller.go

```diff
@@ -92,24 +95,44 @@ func NewTaggingController(
 	cloud cloudprovider.Interface,
 	nodeMonitorPeriod time.Duration,
 	tags map[string]string,
-	resources []string) (*Controller, error) {
+	resources []string,
+	rateLimit float64,
+	burstLimit int) (*Controller, error) {
 
 	awsCloud, ok := cloud.(*awsv1.Cloud)
 	if !ok {
 		err := fmt.Errorf("tagging controller does not support %v provider", cloud.ProviderName())
 		return nil, err
 	}
 
+	var rateLimiter workqueue.RateLimiter
+	var rateLimitEnabled bool
+	if rateLimit > 0.0 && burstLimit > 0 {
```

Comment:
I used these for simplification. Technically, we could support zero values for these flags but they don't seem to make a lot of sense. E.g. 
* Zero rate limit means that nothing is allowed (regardless of the burst value). 
* No rate-limit but bounded burst is hard to reason because if no rate-limit means that the rate-limit is infinite, then the burst doesn't make sense since there'll effectively be no throttling. If we assume no rate-limit as zero rate-limit then we're back to the case above. I didn't want us to make assumptions here so the simplest case seemed to default to zero and use that as an indication for throttling being disabled (we also need that behavior for backwards compatibility).
* Non-zero rate-limit with zero burst might have some uses but then again it'll probably be confusing to the customers because of cases like the above and I am not very sure if this is something that'd be a big deal in particular. Worst case, the burst can be set to 1.

What do you think?

Reply:
You're right. For rate limit, we either want to set it to positive value (to limit steady state) or unset (no limit). Sorry it skipped my mind earlier that this was a client-side limit, so setting no limit wouldn't mean that we're still bounded by the ec2 limit of 5 qps.

But since we're using 0 value here to mean no limit, how can we model a scenario where we want to limit steady-state qps (to let's say 2), while still allow full burst? Might be something we wanna do to handle spikes of new nodes.

Reply:
Actually I realized it's hard to model "allow full burst" without knowing the server-side burst value. We could do something like "burst until server throttles", but that may not be so straight-forward. So let's ignore the above case for now.

Reply:
Yeah, plus "allow full burst" might just mean setting the burst value to a very high number so we could still accomplish this with the current behavior.

Reply:
The problem with setting burst to a very high number means that our client could indeed spam the service (way past their allowed burst). So at that point, the rate-limiting is not practically of any use.

Reply:
Totally, I don't see any practical value for that but I was merely calling out that such a thing is still possible in the current flag implementation in case the customers of this controller want to.

Realistically, the customers of this controller would need to know the server-side rate limitations to properly configure these values anyway.